packages/frontend-2/components/settings/workspaces/members/GuestsTable.vue

```diff
@@ -231,4 +251,20 @@ const onRemoveUser = async () => {
 const toggleMenu = (itemId: string) => {
   showActionsMenu.value[itemId] = !showActionsMenu.value[itemId]
 }
+
+const onUpdateRole = async (newRoleValue: WorkspaceRoles) => {
+  if (!userToModify.value || !newRoleValue) return
+
+  await updateUserRole({
+    userId: userToModify.value.id,
+    role: newRoleValue,
+    workspaceId: props.workspaceId
+  })
+
+  mixpanel.track('Workspace User Role Updated', {
```

Comment:
I think this event will also be sent if `updateUserRole` was executed but not succes, probably need to check the result of this function and see wether it was successful or not

Reply:
Good point, I moved the mixpanel events to management.ts, so we could accurately track this.